### DISCUSSION TITLE
New category request megathread

### DISCUSSION THREAD

**Discussion by triphora** [modrinth/.github#105] [Feature requests]
Please use the following template for new category suggestions:
```
**What category do you suggest?:** Category name

**What is the intended use for this category?:** A brief, one-sentence description of the sorts of projects that could use this category.

**What project type(s) would this category be applicable to?:** Mods, plugins, data packs, shaders, resource packs, modpacks

**Please provide links to at least three *existing* Modrinth projects that could have this category:**
- https://modrinth.com/ ...
- https://modrinth.com/ ...
- https://modrinth.com/ ...

**Any additional context?:** You can remove this question if you want.
```

Here's an example, using an already existing Modrinth category:

<details><summary>Example</summary>

**What category do you suggest?:** Optimization

**What is the intended use for this category?:** Projects that are intended to help with optimizing the game, specifically in regards to FPS, TPS/MSPT, ping, or resource usage.

**What project type(s) would this category be applicable to?:** Mods and modpacks

**Please provide links to at least three *existing* Modrinth projects that could have this category:**
- https://modrinth.com/mod/sodium
- https://modrinth.com/mod/lithium
- https://modrinth.com/modpack/fabulously-optimized

</details>

Feel free to upvote individual requests and use comment threads. Please do not create a new thread if there is already an existing thread for your suggested category.

Please note that we may require multiple upvotes before we consider adding a new category.

**See also:** [the loader request megathread](https://github.com/orgs/modrinth/discussions/106)

**Comment by triphora** [comment]
*Initially requested by @sternschnaube and @Wanesty*

**What category do you suggest?:** Immersive

**What is the intended use for this category?:** For mods that go above and beyond to allow the user to be immersed in the mod, especially in regards to audio

**What project type(s) would this category be applicable to?:** Mods, plugins, data packs

**Please provide links to at least three *existing* Modrinth projects that could have this category:**
- https://modrinth.com/mod/auditory
- https://modrinth.com/mod/physicsmod
- https://modrinth.com/user/YUNGNICKYOUNG

**Any additional context?:** https://github.com/modrinth/knossos/issues/1107

**Reply by wanesty** [reply]
imho an user tag system would be the way to go instead of creating a bunch of categories
this would also allow devs publishing their project to tag them to other compatible stuff, for exemple an AE2 addon having a `AE2` tag
i'm thinking about a "booru" type search (for the searchbar autocomplete part anyways)

**Comment by bangetto** [comment]
**What category do you suggest?:** Vanilla+ or Vanilla-like

**What is the intended use for this category?:** Modpacks that adds features [not just optimization] but keeps the vanilla feel.

**What project type(s) would this category be applicable to?:** Modpacks

**Please provide links to at least three *existing* Modrinth projects that could have this category:**
- https://modrinth.com/modpack/vanilla-twenty
- https://modrinth.com/modpack/landscapes-reimagined/gallery
- https://modrinth.com/modpack/uncharted-horizons/gallery

**Any additional context?:** Currently it's hard to search for these type of modpacks

**Reply by ThexXTURBOXx** [reply]
This could also be a nice category for mods which only slightly change the game without removing the "Vanilla" feeling

**Reply by pac18010** [reply]
I really wish this was available to mods/modpacks/data packs and not just resource packs/shaders

**Reply by FlashSaber007** [reply]
Same was going to post the same suggestion XD

**Comment by ToBinio** [comment]
**What category do you suggest?:** Visual

**What is the intended use for this category?:** A mod that only adds visual improvement/changes without adding new functionality.

**What project type(s) would this category be applicable to?:** Mods (data packs, modpacks)

**Please provide links to at least three *existing* Modrinth projects that could have this category:**
- https://modrinth.com/mod/appleskin
- https://modrinth.com/mod/not-enough-animations
- https://modrinth.com/mod/detail-armor-bar
- https://modrinth.com/mod/physicsmod

**Reply by MIfoodie** [reply]
But doesn't the client side category sort of work as a "visual" category? It's not exactly the same but we won't want a ton of super specific categories.

**Reply by wanesty** [reply]
client-side is not a category to sort feature type

**Reply by Minenash** [reply]
> But doesn't the client side category sort of work as a "visual" category?  

No, because all that filter does is check to see if there is a client side component. It doesn't exclude mods that have server side components (Though negative filters is a near future addition)

>  we won't want a ton of super specific categories

I agree, (sub-cateigories would be helpful for this imo). I don't think Visual is super specific. I think there's a lot of mods that are listed as Utility better fits with this category.

**Reply by wanesty** [reply]
having sub categories would be nice yea

and maybe tags too ?

**Comment by OracleHisty** [comment]
**What category do you suggest?:** Sinytra Connector Support

**What is the intended use for this category?:** Any mods that announce they have built-in support for sinytra connector for fabric mods on forge.

**What project type(s) would this category be applicable to?:** Mods

**Please provide links to at least three *existing* Modrinth projects that could have this category:**
- https://modrinth.com/mod/betterend - compatibility is proven
- https://modrinth.com/mod/terrestria - despite having an unofficial forge port, terrestria is compatible using connector
- https://modrinth.com/mod/sandwichable - again, a mod with proven compatibility.

**Any additional context?:** 
Although this _could_ be treated as a modloader, it probably should not because it has a focus on running fabric mods that use fabric api on forge, Successfully for that matter, but not vice-versa.

https://modrinth.com/mod/spectrum
The creator accepted a PR for support and is interested in implementing support for the future releases.

This pair of "libraries" is useful specifically because Forge profiles WILL NOT show fabric mods that have sinytra support built-in. Usually its up to the individual modder to implement support, so this is a well-deserved category.

**Comment by sternschnaube** [comment]
## **What category do you suggest?** 

Vanilla server-side (The name is expandable - suggestions to outform a better name are highly welcome!)
_____________

## **What is the intended use for this category?** 

An author can mark his project as "Client", "Server", "Client or Server" and "Client and Server". 

Mods that are fully server-side compatible and allow pure vanilla clients to join the server are very popular.

Nevertheless, I notice major inconsistencies in the definition.

Which, in the end, is nobody's fault. Some mod writers are simply not aware of its importance. A solution that results from helping the mod authors themselves and at the end of the community. 

A new category could provide a remedy, which explicitly states whether vanilla clients can connect to a server if they do not have the required mod(s) installed.
_____________
## **What project type would this category be applicable to?** 

Mods.
_____________
## **Please provide links to at least three *existing* Modrinth projects that could have this category:**
- https://modrinth.com/mod/audaki-cart-engine
- https://modrinth.com/mod/all-stackable
- https://modrinth.com/mod/alternate-current

(Absolutely random picks)

## **Additional context:**

It might also be a category that could also be specified for individual version uploads, as older mod versions may not offer server-side support.

I consider this topic to be very important and it would offer great added value. 
I welcome suggestions about this and other opinions. Thank you very much 😊

**Reply by Minenash** [reply]
To be specific, mod authors can mark the two sides with Required/Optional/Unsupported. What you're referring to is "Server required, client optional/unsupported", which iirc is what "Server" means on the project card

If there's any changes needed, it would be related to this system, rather than adding a category.

Though having sidness per version would be cool

**Reply by triphora** [reply]
We already have client-side/server-side compatibility as its own field. Go to your project settings—it lets you mark client and server sidedness right there.

**Comment by 0zadius** [comment]
**What category do you suggest?:** Construction/Building

**What is the intended use for this category?:** I believe that projects such as mods geared towards enhancing building mechanics or modpacks curated specifically with building-centric mods could greatly benefit from being categorized under "Construction."

**What project type(s) would this category be applicable to?:** Mods, plugins, data packs, and modpacks, but particularly modpacks and mods would find this category invaluable. It's frustratingly common to desire a modpack tailored to building but having to extensively research to determine if it includes the desired building mods.

**Please provide links to at least three existing Modrinth projects that could have this category:**

[Chisels & Bits for TFC](https://modrinth.com/mod/chisels-bits-for-tfc) (Mod)
[Axiom](https://modrinth.com/mod/axiom) (Mod)
[Builder's Dreamland](https://modrinth.com/modpack/builders-dreamland) (Modpack)

**Any additional context?:** Just as technical players gravitate towards the Technology category and explorers seek out the Adventure category, builders deserve a streamlined way to sift through mods and modpacks tailored to their craft. The inclusion of a dedicated "Construction" category would empower builders to easily locate projects aligned with their creative endeavors. Sometimes, players crave a modpack encompassing quests, technology, magic, and building components, yet the absence of a Construction/Building category makes filtering for such comprehensive experiences challenging.

**Reply by N0tHorizon** [reply]
I see this as a possibly  but I think it would be to generic and not many mods would have this.

**Comment by pac18010** [comment]
**What category do you suggest?:** HUD or Player HUD

**What is the intended use for this category?:** Anything that affects the player HUD i.e. hunger bar, hearts, armor sprites, bubbles, the hotbar, etc. More specific than the GUI category, which includes any game interface.

**What project type(s) would this category be applicable to?:** Mods, plugins, data packs, shaders, resource packs, modpacks

**Please provide links to at least three *existing* Modrinth projects that could have this category:**
- https://modrinth.com/resourcepack/bread-hunger-bar
- https://modrinth.com/resourcepack/summer-hotbar
- https://modrinth.com/resourcepack/dark-hud/gallery

**Any additional context?:** Would mostly be useful for resource packs, but its possible to make mods/etc that affect the HUD as well

**Reply by Minenash** [reply]
This would be really nice for mods, where everything is kinda forced into utility and decoration

**Reply by N0tHorizon** [reply]
Hud  changes are based around resource  packs which already have this sort filter

**Reply by pac18010** [reply]
https://modrinth.com/mod/sneaksprint-indicator

My rebuttal

**Comment by fabez** [comment]
**What category do you suggest?:**  Distant Horizons Compatible

**What is the intended use for this category?:** To filter out shaders which are compatible with the Distant Horizon's mod as currently this is usually buried in the description.

**What project type(s) would this category be applicable to?:** Shaders

**Please provide links to at least three *existing* Modrinth projects that could have this category:**
https://modrinth.com/shader/complementary-reimagined
https://modrinth.com/shader/insanity-shader
https://modrinth.com/shader/photon-shader
https://modrinth.com/shader/auroras-shaders
https://modrinth.com/shader/bloop-shaders

**Comment by Hexasan** [comment]
**What category do you suggest?:** Addon

**What is the intended use for this category?:** Anything that needs another mod(excluding libraries), resourcepack or datapack to function.

**What project type(s) would this category be applicable to?:** Mods, Data Packs, Resource Packs

**Please provide links to at least three *existing* Modrinth projects that could have this category:**
- https://modrinth.com/resourcepack/extra-xaeros-minimap-icons
- https://modrinth.com/datapack/create-structures
- https://modrinth.com/mod/toomanyorigins

**Any additional context?:** Would be filtered out more rather than filtered in. There is Origin addon/Cobblemon addon/Compat addon clutter especially on the data pack page, it would help with that.

**Reply by OracleHisty** [reply]
What Category do you suggest: Addon

What is the intended use for this category?: Filter Refinement; Extends an existing content project (EXCLUDES LIBRARIES/APIs) and requires them to function

What project type(s) would it apply to?: Mods, Data Packs, Resource Packs, (Server Plugins), EXCLUDES BEDROCK PLATFORM

links to three existing MR projects (see quoted or OP on thread)

Additional Context

CurseForge as a competitor could NOT futureproof the addon category. It was created BEFORE bedrock was externally expandable like it is now. The way Modrinth is structured, would require a dedicated bedrock category to filter OUT before the mutual panics that ensue from ADDING bedrock support and not being clear on what an ADDON is compatible with.

From OP:

> **What category do you suggest?:** Addon
> 
> **What is the intended use for this category?:** Anything that needs another mod(excluding libraries), resourcepack or datapack to function.
> 
> **What project type(s) would this category be applicable to?:** Mods, Data Packs, Resource Packs
> 
> **Please provide links to at least three _existing_ Modrinth projects that could have this category:**
> 
> * https://modrinth.com/resourcepack/extra-xaeros-minimap-icons
> * https://modrinth.com/datapack/create-structures
> * https://modrinth.com/mod/toomanyorigins
> 
> **Any additional context?:** Would be filtered out more rather than filtered in. There is Origin addon/Cobblemon addon/Compat addon clutter especially on the data pack page, it would help with that.


My Commentary: 
I counter-argue using "Addon" as the term, as Bedrock has their respective term for a pack with "Textures and Behaviors" as addons, Instead, if we were to get a dedicated Bedrock Category similarly to the later mentioned plugins Category

> - Mods
>     - Version
>     - Modloader
>     - Tags that already exist
>     - Addon
> - Resource Packs
>     - Shader Packs
>         - Vanilla (Core Shaders)
>         - Optifine(Stolenpatcher)/MCPatcher Support (Mutually exclusive with iris support in filtering only) 
>         - Iris/Occulus Support (Mutually exclusive with Optifine support in filtering only)
>         - Other Shader Libs
>     - Resource Packs
>         - All Already Existing Filters
> - Datapacks
>     - Version
>     - Requires another mod/project for full features *name subject to change*
> - Plugins
>     - Server Type
>     - Plugin Tags
> - Bedrock > see my other suggestions, linked later.
>     - McAddons (Bedrock Exclusive Tag)
>     - Behavior Packs (Bedrock Exclusive Tag)
>     - Texture Packs (Bedrock Exclusive Tag)
>     - McWorlds (Bedrock Exclusive Tag)

**Comment by flikrheist** [comment]
What category do you suggest?: Pluginpacks

**Comment by OracleHisty** [comment]
### NEW TAG: Offsite Monetized

What is the intended use for this category?: Anything that has supporter-related features, also improves project moderation.

What project type(s) would this category be applicable to?: Everything (Primarily Mods and Plugins)

Please provide links to at least three existing Modrinth projects that could have this category:

 Positive (Cosmetic only) Example: [Defile by DrRat](https://modrinth.com/mod/defile)
 Negative (deceptively monetized) Example: [MythicMobs](https://modrinth.com/plugin/mythicmobs)
 Formerly Offisite Monetized Project: [Physics Mod Pro](https://modrinth.com/mod/physicsmod)

### Additional context:

     Monetized Offsite? (my salt with the changes in Mojang TOS and seeing MythicMobs which has multiple externally monetized dependancies popping up on modrinth's website.)
    1. This is primarily an issue with people who limit the functionality of the "Free/Public Build" **and do not explicitly state so in their Project's Page**. If modrinth has plugins, this is a MUST DO to prevent deceptive practices. 
    2. I personally see that having "Cosmetic" features like how DrRat does with their mods from ContentSMP is not against TOS, but MythicMobs breaks that rule to the extreme. It is not even "Pay for a Commercial Liscense if you make x% more than $y dollar value", it locks you to 8 unique instances of its function, but you have to individually purchase all of the dependancies and itself for full access. 
    3. This is absolutely necessary to be a platform with good reputation.

Additional rant context: Return to encouraging open-source-contribution liscenses.

**Anyway, a lot of the search refinement features would need a bit of future-proofing.**

**Reply by Andre601** [reply]
I (sadly) know a project on Modrinth that isn't explicitly mentioning that it is only a "light" version of a premium one... Which is rather misleading for the end user.
People will think they get the full thing and then wonder why there is only some of the advertised features...
Tho the project in my case doesn't even do that, which makes me wonder how it even got passed the review process with how baren the page itself is in terms of mentioning features.

Ideally, I feel like rather than a tag (or in addition to one) should there be a rule, that projects are required to mention in the project page, that they are reduced/"light" version of an otherwise paid product, to avoid general confusion and misleading of people...

---

Tho, I'm confused on where MythicMobs exactly does something bad... I only know it as a free version and am not aware of any limitations it has there...

**Reply by Prospector** [reply]
Please report them if they are advertising premium features that aren't available on Modrinth, the Modrinth description and gallery must be about the project as it exists on Modrinth, not elsewhere.

**Reply by falseresync** [reply]
Just a reminder that selling Minecraft Java mods is against Minecraft EULA - you may receive donations and indirect revenue, but paywalls are illegal. Report any mods that do that not just to Modrinth, but also to the payment platform (be it Patreon, PayPal, or other)

**Reply by OracleHisty** [reply]
> Please report them if they are advertising premium features that aren't available on Modrinth, the Modrinth description and gallery must be about the project as it exists on Modrinth, not elsewhere.



As said in my original post, DrRat does not sell a product with explicitly advantageous mechanics. All the features **are in Befoul** and the only "supporter" feature is customization, which is entirely cosmetic, and does not give any competitive advantage. This does not violate Mojang TOS but may be falsely flagged under Modrinth's agreement by this logic.

I personally believe that if the featureset is limited on the freely distributed one, then it needs to be purged. If cosemtic supporter features are available by supporting the creator, I don't think that it should be removed.

Unfortunately, early access to projects is the currently unmoderated "exploit" in Mojang TOS and a lot of creators seem to have that as the way they circumvent the ruling. (DrRat is one of the few wierd cases where the content is always of high quality, regardless)

Mythicmobs is a harsh offender in EVERY vein of Mojang TOS, and it is not a mod in reality, its a plugin. Their offsite platform also has a lot of ad boxes that bypass adblockers which frequently have inappropriate content on them, on top of being a TOS violator.

I already reported mythicmobs as I am familiar with their monitization practices closely.

**Reply by OracleHisty** [reply]
> I (sadly) know a project on Modrinth that isn't explicitly mentioning that it is only a "light" version of a premium one... Which is rather misleading for the end user. People will think they get the full thing and then wonder why there is only some of the advertised features... Tho the project in my case doesn't even do that, which makes me wonder how it even got passed the review process with how baren the page itself is in terms of mentioning features.

Two suggestions here, try to get in contact with the creator abou them breaking modrinth rules/regs if you like the "product". If they fix it, good. If not, due process.

> Tho, I'm confused on where MythicMobs exactly does something bad... I only know it as a free version and am not aware of any limitations it has there...

They limit the quantity of implemented instances of thier custom mobs (originally to 6, updated recently to 8) on the "lite" version. MythicMobs is 1 of 8 paid dependancies for the collection. Each clocks in at 40$USD to even attain a copy. This does not include any potential later expenses like comissioning custom content be made for these plugins, or purchasing of content packs created by indies FOR the MythicMobs/model engine.

**Reply by falseresync** [reply]
> "supporter" feature is customization

Cosmetics and other non-intrusive monetization is only allowed for servers, not for mods

**Comment by ghost** [comment]
**What category do you suggest?:** Server

**What is the intended use for this category?:** Modpacks designed for use in specific servers.

**What project type(s) would this category be applicable to?:** Modpacks

**Please provide links to at least three *existing* Modrinth projects that could have this category:**
- https://modrinth.com/modpack/nybblecraft
- https://modrinth.com/modpack/zeldacraft-server-pack
- https://modrinth.com/modpack/souloforigins

**Any additional context?:** We already have the Multiplayer category, but that seems to also be for optimization packs designed for vanilla servers. This would only be for modpacks for specific modded servers.
